src/Adapters/EventDispatcherAdapter.php

```diff
@@ -44,10 +44,13 @@ abstract class EventDispatcherAdapter implements SymfonyDispatcher
      *
      * @api
      */
-    public function dispatch($eventName, Event $event = null)
+    public function dispatch(object $event, ?string $eventName = null): object
     {
-        $this->laravelDispatcher->fire($eventName, $event);
-        return $this->symfonyDispatcher->dispatch($eventName, $event);
+        if ($eventName !== null) {
+            // TODO: Someone who understands events under the hood more should look into this
```

Comment:
What do you need to know?

Within https://github.com/php-tmdb/api/blob/4.1/lib/Tmdb/Event/Listener/RequestListener.php#L70 the request / response handling is done. Most of the other listeners are more "helpers" to add information to the request, or divert the response to a cached instance. 

See how this is configured in symfony; 
- https://github.com/php-tmdb/symfony/blob/5.0/Resources/config/services.xml#L24
- https://github.com/php-tmdb/symfony/blob/5.0/DependencyInjection/CompilerPass/ConfigurationPass.php#L35